src/client/firebase/auth.js

```diff
@@ -0,0 +1,76 @@
+import { auth } from './index';
+
+const FIREBASE_ERROR_CODES = {
+  WEAK_PASSWORD: 'auth/weak-password',
+  WRONG_PASSWORD: 'auth/wrong-password',
+  INVALID_EMAIL: 'auth/invalid-email',
+  USER_NOT_FOUND: 'auth/user-not-found',
+};
+
+/**
+ *
+ * @param {email, password} (sign in user)
+ */
+export async function signIn({ email, password }) {
+  console.log('he');
+  try {
+    console.log({ email, password });
+    await auth.signInWithEmailAndPassword(email, password);
+  } catch (error) {
+    // Handle Errors here.
+    const errorCode = error.code;
+    const errorMessage = error.message;
+    // [START_EXCLUDE]
+    if (errorCode === FIREBASE_ERROR_CODES.WRONG_PASSWORD) {
+      alert('Wrong password.');
+    } else {
+      alert(errorMessage);
+    }
+    console.log(error);
+    // [END_EXCLUDE]
+  }
+}
+
+export async function signUp({ email, password }) {
+  try {
+    await auth.createUserWithEmailAndPassword(email, password);
+    return true;
```

Comment:
So i guess the signup function returns a promise that resolves with true in this case. Do we then want to return false in the catch? Why are you returning true here?

Reply:
I could also return what the result of the function returns 
```javascript
createUserWithEmailAndPassword(
      email: string,
      password: string
    ): Promise<firebase.auth.UserCredential>;
```
but I am not interested in the credentials so I just use `true` and if the function fails it throws.

and if we don't return anything it is `undefined`. but we can explicitly `return false` if you prefer that